service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwin.java

```diff
@@ -504,6 +505,11 @@ private DeviceTwinDevice jsonToDeviceTwinDevice(String json) throws IOException
         deviceTwinDevice.setConnectionState(twinState.getConnectionState());
         deviceTwinDevice.setConfigurations(twinState.getConfigurations());
 
+        if (twinState.getModelId() != null && !twinState.getModelId().isEmpty())
+        {
+            deviceTwinDevice.setModelId(twinState.getModelId());
```

Comment:
unrelated to this PR, but why did we name this `DeviceTwinDevice`? Is it to separate a device twin from a module twin? Is it public? Can we rename it?

Reply:
synced offline on this.